File: drills/ts/harnesses/nc29_car_fleet.ts
import * as solution from "./solution";

const show = (v: unknown) => JSON.stringify(v) ?? "undefined";

export function run(): [string, string, string][] {
  if (typeof solution.carFleet !== "function") throw new Error("__signature_mismatch__");
  return [
    ["carFleet(12, [10, 8, 0, 5, 3], [2, 4, 1, 1, 3])", show(3), show(solution.carFleet(12, [10, 8, 0, 5, 3], [2, 4, 1, 1, 3]))],
    ["carFleet(10, [3], [3])", show(1), show(solution.carFleet(10, [3], [3]))],
    ["carFleet(100, [0, 2, 4], [4, 2, 1])", show(1), show(solution.carFleet(100, [0, 2, 4], [4, 2, 1]))],
    ["carFleet(10, [6, 8], [3, 2])", show(2), show(solution.carFleet(10, [6, 8], [3, 2]))],
    ["carFleet(10, [], [])", show(0), show(solution.carFleet(10, [], []))],
    ["carFleet(10, [0, 4, 2], [2, 1, 3])", show(1), show(solution.carFleet(10, [0, 4, 2], [2, 1, 3]))],
  ];
}

File: drills/go/harnesses/nc29_car_fleet.go
package main

func main() {
	run(func() []testCase {
		return []testCase{
			tc("carFleet(12, [10,8,0,5,3], [2,4,1,1,3])", 3, carFleet(12, []int{10, 8, 0, 5, 3}, []int{2, 4, 1, 1, 3})),
			tc("carFleet(10, [3], [3])", 1, carFleet(10, []int{3}, []int{3})),
			tc("carFleet(100, [0,2,4], [4,2,1])", 1, carFleet(100, []int{0, 2, 4}, []int{4, 2, 1})),
			tc("carFleet(10, [6,8], [3,2])", 2, carFleet(10, []int{6, 8}, []int{3, 2})),
		}
	})
}
